### PR TITLE
MIC-524: Cypress/Jest code coverage report

### PR DESCRIPTION
**Pull request by JackReeveMoJ**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -100,14 +104,15 @@ jobs:
           name: Wait for node app to start
           command: sleep 5
       - run:
-          name: integration tests
-          command: npm run int-test
+          name: Integration tests
+          command: circleci tests glob "**/*.cy.ts" | circleci tests run --command="xargs npx cypress run --reporter cypress-circleci-reporter --spec" --verbose --split-by=timings
```
Comment (ReedSoftware) [inline .circleci/config.yml:108]: How come we're using this special command now instead of `npm run int-test`?

File: server/app.ts [hunk 2]
```diff
@@ -3,6 +3,8 @@ import express from 'express'
 import dpsComponents from '@ministryofjustice/hmpps-connect-dps-components'
 
 import * as Sentry from '@sentry/node'
+// @ts-expect-error Import untyped middleware for cypress coverage
```
Comment (ReedSoftware) [inline server/app.ts:6]: There is a DT implementation apparently https://www.npmjs.com/package/@types/cypress__code-coverage - worth a try?

File: .circleci/config.yml (hunk 1 above)
Comment (JackReeveMoJ) [reply .circleci/config.yml:108]: This is for CircleCI to split the tests up across its different instances. DIdn't want to include this in the package.json as its specific to CircleCI

File: server/app.ts (hunk 2 above)
Comment (JackReeveMoJ) [reply server/app.ts:6]: We do have this in the package.json, but it still complains

File: .circleci/config.yml (hunk 1 above)
Comment (ReedSoftware) [reply .circleci/config.yml:108]: Ah right and there's no way to encode this into the config instead and still run this normally via `npm run int-test`? If not np

File: server/app.ts (hunk 2 above)
Comment (ReedSoftware) [reply server/app.ts:6]: What a pain. Ok, no worries then.

**Review by ReedSoftware — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (JackReeveMoJ) [reply .circleci/config.yml:108]: Not that I'm aware of, I believe this code needs to run within the context of the pipeline and we let circleCI do the splitting

**Pull request merged**